tests/integration/integrations/conftest.py

```diff
@@ -53,55 +224,52 @@ async def deploy_and_configure_minio(ops_test: OpsTest, get_unit_ips) -> None:
         mc_client.make_bucket("tempo")
 
     # configure s3-integrator
-    s3_integrator_app: Application = ops_test.model.applications["s3-integrator"]
-    s3_integrator_leader: Unit = s3_integrator_app.units[0]
-
-    await s3_integrator_app.set_config(
+    juju.config(
+        "s3-integrator",
         {
-            "endpoint": f"minio-0.minio-endpoints.{ops_test.model.name}.svc.cluster.local:9000",
+            "endpoint": f"minio-0.minio-endpoints.{juju.status().model.name}.svc.cluster.local:9000",
             "bucket": "tempo",
-        }
+        },
     )
 
-    action = await s3_integrator_leader.run_action("sync-s3-credentials", **config)
-    action_result = await action.wait()
-    assert action_result.status == "completed"
+    task = juju.run("s3-integrator/0", "sync-s3-credentials", config)
+    assert task.status == "completed"
 
 
-@pytest_asyncio.fixture(scope="module", name="tempo_app")
-async def deploy_tempo_cluster(ops_test: OpsTest, get_unit_ips):
+@pytest.fixture(scope="module", name="tempo_app")
+def deploy_tempo_cluster(
+    juju: jubilant.Juju,
+):
     """Deploys tempo in its HA version together with minio and s3-integrator."""
     tempo_app = "tempo"
     worker_app = "tempo-worker"
     tempo_worker_charm_url, worker_channel = "tempo-worker-k8s", "edge"
     tempo_coordinator_charm_url, coordinator_channel = "tempo-coordinator-k8s", "edge"
-    await ops_test.model.deploy(
+    juju.deploy(
         tempo_worker_charm_url,
-        application_name=worker_app,
+        app=worker_app,
         channel=worker_channel,
         trust=True,
     )
-    app = await ops_test.model.deploy(
+    juju.deploy(
         tempo_coordinator_charm_url,
-        application_name=tempo_app,
+        app=tempo_app,
         channel=coordinator_channel,
         trust=True,
     )
-    await ops_test.model.deploy("s3-integrator", channel="edge")
-    await ops_test.model.integrate(tempo_app + ":s3", "s3-integrator" + ":s3-credentials")
-    await ops_test.model.integrate(tempo_app + ":tempo-cluster", worker_app + ":tempo-cluster")
-    await deploy_and_configure_minio(ops_test, get_unit_ips)
-
-    async with ops_test.fast_forward():
-        await ops_test.model.wait_for_idle(
-            apps=[tempo_app, worker_app, "s3-integrator"],
-            status="active",
-            timeout=2000,
-            idle_period=30,
-            # TODO: remove when https://github.com/canonical/tempo-coordinator-k8s-operator/issues/90 is fixed
-            raise_on_error=False,
-        )
-    return app
+    juju.deploy(
+        "s3-integrator",
+        channel="edge",
+    )
+    juju.integrate(tempo_app + ":s3", "s3-integrator" + ":s3-credentials")
+    juju.integrate(tempo_app + ":tempo-cluster", worker_app + ":tempo-cluster")
```

Comment:
Should we prefer f-strings than string concatenation?
```suggestion
    juju.integrate(f"{tempo_app}:s3", "s3-integrator:s3-credentials")
    juju.integrate(f"{tempo_app}:tempo-cluster", f"{worker_app}:tempo-cluster")
```
It is best practice to prefer f-strings above string concatenations.